dangerzone/gui/updater.py

```diff
@@ -0,0 +1,225 @@
+    )
+
+    def __init__(self, dangerzone: DangerzoneGui):
+        super().__init__()
+        self.dangerzone = dangerzone
+
+    ###########
+    # Helpers for updater settings
+    #
+    # These helpers make it easy to retrieve specific updater-related settings, as well
+    # as save the settings file, only when necessary.
+
+    def _save_me_maybe(self, key: str, val: Any) -> None:
+        """Check the new settings value with the old one, and optionally save it."""
+        old_val = self.dangerzone.settings.updater_get(key)
+        if val == old_val:
+            return
+
+        self.dangerzone.settings.updater_set(key, val)
+        self.dangerzone.settings.save()
+
+    @property
+    def first_run(self) -> bool:
+        return self.dangerzone.settings.updater_get("first_run")
+
+    @first_run.setter
+    def first_run(self, val: bool) -> None:
+        self._save_me_maybe("first_run", val)
+
+    @property
+    def check(self) -> Optional[bool]:
+        return self.dangerzone.settings.updater_get("check")
+
+    @check.setter
+    def check(self, val: bool) -> None:
+        self._save_me_maybe("check", val)
+
+    def prompt_for_checks(self) -> bool:
+        """Ask the user if they want to be informed about Dangerzone updates."""
+        log.debug("Prompting the user for update checks")
+        # TODO: Render HTML links properly
+        # FIXME: Handle the case where a user clicks on "X", instead of explicity making
+        # a choice. We should probably ask them again on the next run.
+        check = Alert(
+            self.dangerzone,
+            message=MSG_CONFIRM_UPDATE_CHECKS,
+            ok_text="Yes",
+            cancel_text="No",
+        ).launch()
+        return bool(check)
+
+    def should_check_for_updates(self) -> bool:
+        """Determine if we can check for updates based on settings and user prefs."""
+        log.debug("Checking platform type")
+        # TODO: Disable updates for Homebrew installations.
+        # TODO: Add a flag that will force-check for updates, so that we can test it on
+        # Linux. Once we have this flag, uncomment the following lines.
+        # if platform.system() == "Linux" and os.environ():
+        #     log.debug("Running on Linux, disabling updates")
+        #     self.check = False
+        #     return False
+
+        log.debug("Checking if first run of Dangerzone")
+        if self.first_run:
+            log.debug("Dangerzone is running for the first time, updates are stalled")
+            self.first_run = False
+            return False
+
+        log.debug("Checking if user has already expressed their preference")
+        if self.check is None:
+            log.debug("User has not been asked yet for update checks")
+            self.check = self.prompt_for_checks()
+            # TODO: Should we always check for updates in the third run, or can we check
+            # them immediately, if the user wants to enable them?
```

Comment:
Ok, cool. My next concern is if this would apply when the user clicks on "Check for updates". Note that we currently don't trigger update checks when the user checks this box.

If we did, we'd need to have some logic in place to avoid multiple menu entries and threads. I prefer tackling this issue in a different iteration, if you're ok with it.

Reply:
Sure.